File: Array/1828.Queries on Number of Points Inside a Circle/Solution.py
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        res = list()
        for i, v1 in enumerate(groupSizes):
            if v1 == -1:
                continue
            temp = list()
            for j, v2 in enumerate(groupSizes):
                if len(temp) < v1 and v2 == v1:
                    temp.append(j)
                    groupSizes[j] = -1
            res.append(temp)
        return res

File: Array/1828.Queries on Number of Points Inside a Circle/Solution.java
class Solution {
    public List<List<Integer>> groupThePeople(int[] groupSizes) {
        List<List<Integer>> res = new ArrayList<>();
        int m = groupSizes.length;
        for (int i = 0; i < m; i++) {
            if (groupSizes[i] == -1) {
                continue;
            }
            int cur = groupSizes[i];
            List<Integer> temp = new ArrayList<>();
            for (int j = 0; j < m; j++) {
                if (temp.size() >= cur) {
                    j = m;
                    continue;
                }
                if (groupSizes[j] == cur) {
                    temp.add(j);
                    groupSizes[j] = -1;
                }
            }
            res.add(temp);
        }
        return res;
    }

}